### PR TITLE
Use namespace selectors to ignore namespaces instead of creating label patches

### LINKED ISSUE #6: Network Policies before ResourceLocker namespaces labels applied
We had an timing issue occurred because of the network policies have been applied, before the namespace labels
```
    network-policies.syn.tools/no-defaults: "true"
    network-policies.syn.tools/purge-defaults: "true"
```
have been three. This caused several APIs haven't been accessible:
```
$ oc api-resources
error: unable to retrieve the complete list of server APIs: apps.openshift.io/v1: the server is currently unable to handle the request, authorization.openshift.io/v1: the server is currently unable to handle the request, image.openshift.io/v1: the server is currently unable to handle the request, metrics.k8s.io/v1beta1: the server is currently unable to handle the request, oauth.openshift.io/v1: the server is currently unable to handle the request, project.openshift.io/v1: the server is currently unable to handle the request, quota.openshift.io/v1: the server is currently unable to handle the request, route.openshift.io/v1: the server is currently unable to handle the request, template.openshift.io/v1: the server is currently unable to handle the request
```

## Steps to Reproduce the Problem

  1. Set up a fresh cluster
  1. Apply the component networkpolicy

## Actual Behavior
There is a timing issue, that Espejo might be faster than the ResourceLocker causing this issue.

## Expected Behavior
There is a clear dependency, that only if the labels have been set, Espejo is going to create the network policies.

### PR DESCRIPTION
**Pull request by bastjan**
Fixes #6.

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [ ] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Link this PR to related issues.

### CODE REVIEW

File: tests/golden/defaults/networkpolicy/networkpolicy/05_purge_defaults.yaml [hunk 1]
```diff
@@ -7,8 +7,32 @@ metadata:
...
+  deleteItems:
+    - apiVersion: networking.k8s.io/v1
+      kind: NetworkPolicy
+      name: allow-from-same-namespace
+    - apiVersion: networking.k8s.io/v1
+      kind: NetworkPolicy
+      name: allow-from-other-namespaces
+  namespaceSelector:
+    matchNames:
+      - my-ignored-namespace
```
[the remark this answers is not in the document]
Comment (bastjan) [reply tests/golden/defaults/networkpolicy/networkpolicy/05_purge_defaults.yaml:23]: It's for purging the default. 

The namespace is ignored here for applying defaults: https://github.com/projectsyn/component-networkpolicy/pull/11/files#diff-52f54b0ffc3fec8622d3cca770eab37947c23036da3752e19ac5e29bf1a1179bR15

[the remark this answers is not in the document]
Comment (ccremer) [reply tests/golden/defaults/networkpolicy/networkpolicy/05_purge_defaults.yaml:23]: ah, now I get it. You are trying to cleanup the old stuff via this component, amirite?

[the remark this answers is not in the document]
Comment (bastjan) [reply tests/golden/defaults/networkpolicy/networkpolicy/05_purge_defaults.yaml:23]: Yes i think this was the goal. Ignored namespaces were purged before this MR so I kept this functionality.

**Review by ccremer — APPROVED** [review]
Generally LGTM. But isn't this a breaking change? I guess users have to reconfigure the component?

**Comment by bastjan** [comment]
> Generally LGTM. But isn't this a breaking change? I guess users have to reconfigure the component?

Nothing in the config changed and labels and their functions stayed the same. I'll still have to test it more but nothing should break.

**Pull request merged**